share/src/main/java/org/apache/kafka/server/share/SharePartitionKey.java

```diff
@@ -72,6 +72,21 @@ public static SharePartitionKey getInstance(String groupId, TopicIdPartition top
      * @return object representing SharePartitionKey
      */
     public static SharePartitionKey getInstance(String key) {
+        validate(key);
+        String[] tokens = key.split(":");
+        return new SharePartitionKey(
+                tokens[0],
+                Uuid.fromString(tokens[1]),
+                Integer.parseInt(tokens[2])
+        );
+    }
+
+    /**
+     * Validates whether the String argument has a valid SharePartitionKey format - groupId:topicId:partition
+     * @param key - String in format groupId:topicId:partition
+     * @exception IllegalArgumentException if the key is empty or has invalid format
```

Comment:
In Kafka I have seen usage of @throws instead of @exception, with an exception in only one class. Though they are synonyms in java but shall we follw the general way?